model/inline/src/main/scala/aqua/model/inline/raw/CollectionRawInliner.scala

```diff
@@ -15,16 +25,13 @@ object CollectionRawInliner extends RawInliner[CollectionRaw] {
     propertiesAllowed: Boolean
   ): State[S, (ValueModel, Inline)] =
     for {
-      streamName <- Mangler[S].findAndForbidName(
-        (
-          raw.boxType match {
-            case _: StreamType => "stream"
-            case _: CanonStreamType => "canon_stream"
-            case _: ArrayType => "array"
-            case _: OptionType => "option"
-          }
-        ) + "-inline"
-      )
+      streamName <-
+        raw.boxType match {
+          case _: StreamType => raw.assignToName.map(s => State.pure(s)).getOrElse(Mangler[S].findAndForbidName("stream-inline"))
```

Comment:
This `CollectionRawInliner` has a very specific callsite where you already have this `assignToName` in scope, I suppose